### ISSUE TITLE
Keras 2.3.0 last release to support multi-backend Keras

### ISSUE THREAD

**Issue by cfox570** [plaidml/plaidml#586]
From KERAS.IO

_**DOES THIS PREVENT PLAIDML FROM SUPPORTING KERAS IN THE FUTURE?**_

Multi-backend Keras and tf.keras:

At this time, we recommend that Keras users who use multi-backend Keras with the TensorFlow backend switch to tf.keras in TensorFlow 2.0. tf.keras is better maintained and has better integration with TensorFlow features (eager execution, distribution support and other).

Keras 2.2.5 was the last release of Keras implementing the 2.2.* API. It was the last release to only support TensorFlow 1 (as well as Theano and CNTK).

The current release is Keras 2.3.0, which makes significant API changes and add support for TensorFlow 2.0. **The 2.3.0 release will be the last major release of multi-backend Keras. Multi-backend Keras is superseded by tf.keras.**

Bugs present in multi-backend Keras will only be fixed until April 2020 (as part of minor releases).

For more information about the future of Keras, see the Keras meeting notes.

**Comment by AGenchev** [comment]
Yes, I also saw this. Interesting whether the developer (isn't it Intel) will decide to support own version of Keras (if not prohibited by a license) with the same front-end functions called by analogy *plaidml.keras* or will develop their own high-level functions. I'm new here and I like Keras front, because it describes the NN structure in readable way. I like plaidml, because I have Intel GPU on laptop and while  my nvidia has 2GB (training OOMs), Intel write on their site they can use up to 32GB system RAM ...

**Comment by dgkutnic** [comment]
Hi, thanks for using PlaidML and expressing your concern about the ramifications of the Keras statement regarding multi-backend support. One of our contributors (@SleepProgger) had mentioned this to us on our Slack channel recently too.

We are currently investigating the effort and feasibility of supporting Keras once multi-backend support is deprecated. We hope to have a roadmap update concerning this issue soon, and I'll leave this issue open for discussion in the meantime.

**Comment by bhack** [comment]
You can try to follow these threads:
https://github.com/keras-team/governance/issues/2
https://github.com/keras-team/keras-contrib/issues/519

**Comment by Ian-Zhabin** [comment]
@dgkutnic Hi, Any updates for "roadmap update "?

**Comment by akuz** [comment]
So for now, if I want to use PlaidML+Keras, I should **not** be using tf.keras, right?

**Comment by artyom-beilis** [comment]
Please elaborate more about this decision.

One of the reasons Keras was interesting is multi-backend support. So far it was only options to use for example AMD GPU on Mac OS X or on Windows is to use PlaidML + Keras.

It is only option to use Open Source cross platform training backend (based on OpenCL)

So reasoning behind removing multi-backend support isn't clear at all.

**Comment by artyom-beilis** [comment]
Any updates on future of Keras + PlaidML integration?

**Comment by flaub** [comment]
Since keras dropped multi-backend support, the future of integration with `tf.keras` is to actually integrate plaidml with Tensorflow. This is ongoing work.

**Comment by AGenchev** [comment]
Some people come to plaidml, because TF targeted Nvidia GPU while PlaidML wants to support others via OpenCL. If TF has become multi-backend, this makes sense.
